src/main/java/app/persistence/OrderMapper.java

```diff
@@ -0,0 +1,266 @@
+                LocalDateTime timestamp = rs.getTimestamp("order_timestamp").toLocalDateTime();
+                order = new Order(orderId, userId, getAllOrderItems(orderId, connectionPool), orderStatus, timestamp);
+            } else {
+                throw new DatabaseException("The order id doesn't exist");
+            }
+
+        } catch (SQLException e) {
+            throw new DatabaseException("Database Error", e.getMessage());
+        }
+        return order;
+    }
+
+
+    private static List<OrderItem> getAllOrderItems(int orderId, ConnectionPool connectionPool) throws SQLException, DatabaseException {
+        List<OrderItem> orderItems = new ArrayList<>();
+
+        String sql = "SELECT order_item_id, cupcake_bottom_id, cupcake_top_id, order_item_quantity FROM order_items WHERE order_id = ?";
+
+        try (
+                Connection connection = connectionPool.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql)
+        ) {
+            ps.setInt(1, orderId);
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                int orderItemId = rs.getInt("order_item_id");
+                int cupcakeBottomId = rs.getInt("cupcake_bottom_id");
+                int cupcakeTopId = rs.getInt("cupcake_top_id");
+                int orderItemQuantity = rs.getInt("order_item_quantity");
+                orderItems.add(new OrderItem(orderItemId, orderId, CupcakeMapper.getCupcakePartById(cupcakeTopId, connectionPool, CupcakePart.Type.TOP), CupcakeMapper.getCupcakePartById(cupcakeBottomId, connectionPool, CupcakePart.Type.BOTTOM), orderItemQuantity));
+            }
+
+        } catch (SQLException e) {
+            throw new DatabaseException("Failed to get all order items");
+        }
+        return orderItems;
+    }
+
+    private static void createOrderItem(OrderItem orderItem, ConnectionPool connectionPool) throws DatabaseException {
+
+        String sql = "INSERT INTO order_items (order_id, cupcake_bottom_id, cupcake_top_id, order_item_quantity) values (?,?,?,?)";
+
+        try (
+                Connection connection = connectionPool.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql)
+        ) {
+            ps.setInt(1, orderItem.getOrderId());
+            ps.setInt(2, orderItem.getCupcakeBottom().getCupcakePartId());
+            ps.setInt(3, orderItem.getCupcakeTop().getCupcakePartId());
+            ps.setInt(4, orderItem.getQuantity());
+            int rowsAffected = ps.executeUpdate();
+            if (rowsAffected != 1) {
+                throw new DatabaseException("Error could not insert order item");
+            }
+        } catch (SQLException e) {
+            throw new DatabaseException("Error in DB connection", e.getMessage());
+        }
+    }
+
+    private static void deleteOrderItem(int orderId, ConnectionPool connectionPool) throws DatabaseException {
```

Comment:
Rename to deleteOrderItems with an `s` to make it clear multiple order items are being deleted at once